core/src/main/java/org/apache/accumulo/fate/zookeeper/ZooLock.java

```diff
@@ -60,6 +60,19 @@ public String toString() {
 
   }
 
+  public static class ZooLockPath {
+    private final String path;
+
+    public ZooLockPath(String path) {
+      this.path = path;
+    }
+
+    @Override
+    public String toString() {
+      return this.path;
+    }
+  }
+
```

Comment:
Just a general observation - would it improve readability if in addition to toString() that there was a getter (getPath())? 

Also, did you notice places where the node name was necessary?  If so, then would things be cleaner if there was a getNode() or something that did the parsing?  Might save a few lines here and there - but would gather that code into a single place.

Reply:
> Just a general observation - would it improve readability if in addition to toString() that there was a getter (getPath())?

I am not sure if it would improve readability. It could increase robustness but I do not think it benefits that much. 

> Also, did you notice places where the node name was necessary? If so, then would things be cleaner if there was a getNode() or something that did the parsing? Might save a few lines here and there - but would gather that code into a single place.

Most replacements were just for the string with this new path object (and then having toString when necessary). I did not notice anything about node name while making these changes.

Reply:
@EdColeman These shouldn't be thought of as an object that *has* a path... rather, they *are* path types. So, adding `getPath()` to these would make them less concise and somewhat confusing (a path type with a getter for a path type is confusing to me, anyway). I think having a proper `toString()` here is sufficient.